src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -18,30 +18,48 @@ namespace AutoRest.CSharp.Mgmt.Output
 {
     internal class MgmtObjectType : SchemaObjectType
     {
-        private ObjectTypeProperty[]? _myProperties;
+        private IReadOnlyList<ObjectTypeProperty>? _myProperties;
 
         public MgmtObjectType(InputModelType inputModel, SerializableObjectType? defaultDerivedType = null)
-            : base(inputModel, inputModel.Namespace ?? MgmtContext.Context.DefaultNamespace, MgmtContext.TypeFactory, MgmtContext.Context.SourceInputModel, defaultDerivedType)
+            : base(inputModel, Configuration.Namespace, MgmtContext.TypeFactory, MgmtContext.Context.SourceInputModel, defaultDerivedType)
         {
         }
 
         protected virtual bool IsResourceType => false;
         private string? _defaultName;
         protected override string DefaultName => _defaultName ??= GetDefaultName(InputModel, IsResourceType);
         private string? _defaultNamespace;
-        protected override string DefaultNamespace => _defaultNamespace ??= GetDefaultNamespace(MgmtContext.Context, InputModel, IsResourceType);
+        protected override string DefaultNamespace => _defaultNamespace ??= GetDefaultNamespace(MgmtContext.Context, IsResourceType);
 
-        internal ObjectTypeProperty[] MyProperties => _myProperties ??= InputModel.Properties.Select(CreateProperty).ToArray();
+        internal IReadOnlyList<ObjectTypeProperty> MyProperties => _myProperties ??= BuildMyProperties();
+
+        private IReadOnlyList<ObjectTypeProperty> BuildMyProperties()
+        {
+            if (InputModel.IsUnknownDiscriminatorModel)
+            {
+                return InputModel.Properties.Select(CreateProperty).ToArray();
+            }
+
+            var result = new List<ObjectTypeProperty>();
+
+            // For TypeSpec input, we need to add all the properties from the base types
+            // For swagger input, the base type properties are already included in the properties list, we will just skip them
+            var propertyNames = InputModel.Properties.Select(p => p.Name).ToHashSet();
```

Comment:
we should be able to remove this when https://github.com/Azure/autorest.csharp/pull/4797 is merged